lib/bugsnag/integrations/railtie.rb

```diff
@@ -91,10 +91,19 @@ def event_subscription(event)
         filtered_data = data.slice(*event[:allowed_data])
         filtered_data[:event_name] = event[:id]
         filtered_data[:event_id] = event_id
-        if event[:id] == "sql.active_record" && data.key?(:binds)
-          binds = data[:binds].each_with_object({}) { |bind, output| output[bind.name] = '?' if defined?(bind.name) }
-          filtered_data[:binds] = JSON.dump(binds) unless binds.empty?
+
+        if event[:id] == "sql.active_record"
+          if data.key?(:binds)
+            binds = data[:binds].each_with_object({}) { |bind, output| output[bind.name] = '?' if defined?(bind.name) }
+            filtered_data[:binds] = JSON.dump(binds) unless binds.empty?
+          end
+
+          if data.key?(:connection) && !data.key?(:connection_id)
```

Comment:
Might be worth annotating `connection_id` here or in the `allowed_data` list to show its for older versions.